### ISSUE TITLE
Timestamp parsing fails in certain locales, preventing correct audio condensation

### ISSUE THREAD

**Issue by Daja177** [Ajatt-Tools/impd#13]
## Description
The [`parse_speech_fragments`](https://github.com/Ajatt-Tools/impd/blob/68d6f80c688a3f84483382559bdc402b9bbb3cec/impd#L376) function incorrectly parses timestamps on certain distros using certain locales, such as the Swedish locale (`sv_SE.UTF-8`) on Ubuntu-based distribution Tuxedo OS. This causes the function to output timestamps incorrectly (e.g., `00:00:01,640 --> 00:00:03,517` becomes `1,3` instead of `1.64,3.517`). As a result, users with these locales are unable to use the tool to condense audio, as the output format is incompatible with other processing stages.

## Steps to Reproduce
1. Set the numeric locale to a problematic one using `LC_NUMERIC=sv_SE.UTF-8` on Tuxedo OS or perhaps another Ubuntu-based distribution.
2. Condense video with subtitles using either `impd add` or `impd condense`.
3. Observe how condensing fails due to attempting to process chunks from e.g. timestamp `0` to `8,3,2` rather than from `1.44` to `3.717`.

## Potential Solution
Enforce a consistent numeric locale by setting `LC_NUMERIC=C` for the `awk` command within the function. This approach ensures that numeric formatting remains independent of system locale settings:
```bash
parse_speech_fragments() {
    LC_NUMERIC=C awk -F' --> ' -vPADDING="$padding" -vSRT_TIMING_PATTERN="$srt_timing_pattern" '
    ...
}'
```
This setting has resolved the issue in testing, allowing the function to output the correct format on my system with the problematic locale.

## Environment
* OS: Tuxedo OS (Ubuntu-based distribution)
* Locale: `sv_SE.UTF-8` (Swedish)

**Comment by tatsumoto-ren** [comment]
I tried running `LC_NUMERIC=sv_SE.UTF-8 impd add some-file-path`, but couldn't reproduce the issue. My distro is not based on ubuntu though.

`parse_speech_fragments()` doesn't call any OS-specific functions, it just parses SRT timings. Why do you think the issue is caused by the locale?

**Comment by Daja177** [comment]
I specified the distro both in the description and steps to reproduce because I haven't experienced this issue with any other I've been using prior, and during testing I noticed the issue didn't occur in Arch through Distrobox even with the same locale settings.

As for why I think the issue is caused by the locale: as described in the issue, on the affected system, changing the value of environment variable `LC_NUMERIC` from `sv_SE.UTF-8` to `C`, `fr_FR.UTF-8` or `en_US.UTF-8` fixed the issue and made it parse correctly. Changing it back to `sv_SE.UTF-8` afterwards causes the issue to occur again.
